test/e2e/cluster_labels_e2e/cluster_labels_test.go

```diff
@@ -0,0 +1,222 @@
+					offloadingv1alpha1.EnforceSameNameMappingStrategyType,
+					offloadingv1alpha1.LocalAndRemotePodOffloadingStrategyType,
+					testutils.GetClusterSelector())
+			}, timeout, interval).Should(Succeed())
+
+			By(fmt.Sprintf(" 3 - Getting the virtual nodes in the cluster '%d'", localIndex))
+			virtualNodesList := &corev1.NodeList{}
+			Eventually(func() error {
+				return testContext.Clusters[localIndex].ControllerClient.List(ctx, virtualNodesList,
+					client.MatchingLabels{liqoconst.TypeLabel: liqoconst.TypeNode})
+			}, timeout, interval).Should(Succeed())
+			Expect(len(virtualNodesList.Items)).To(Equal(testconsts.NumberOfTestClusters - 1))
+
+			By(" 4 - Checking the remote clusters on which the remote namespaces are created")
+			for i := range virtualNodesList.Items {
+				match, err := k8shelper.MatchNodeSelectorTerms(&virtualNodesList.Items[i], testutils.GetClusterSelector())
+				Expect(err).To(Succeed())
+				remoteClusterID := virtualNodesList.Items[i].Annotations[liqoconst.RemoteClusterID]
+				if match {
+					// Check if the remote namespace is correctly created.
+					By(fmt.Sprintf(" 5 - Checking if a remote namespace is correctly created inside cluster '%s'", remoteClusterID))
+					Eventually(func() error {
+						namespace := &corev1.Namespace{}
+						if err = testContext.ClustersClients[remoteClusterID].Get(ctx,
+							types.NamespacedName{Name: testNamespaceName}, namespace); err != nil {
+							return err
+						}
+						value, ok := namespace.Annotations[liqoconst.RemoteNamespaceAnnotationKey]
+						if !ok {
+							return fmt.Errorf("the annotation '%s' is not present", liqoconst.RemoteNamespaceAnnotationKey)
+						}
+						if value != testContext.Clusters[localIndex].ClusterID {
+							return fmt.Errorf("the value of the annotation is not correct: '%s'", value)
+						}
+						return nil
+					}, longTimeout, interval).Should(Succeed())
+
+				} else {
+					// Check if the remote namespace does not exists.
+					By(fmt.Sprintf(" 5 - Checking that no remote namespace is created inside cluster '%s'", remoteClusterID))
+					Consistently(func() metav1.StatusReason {
+						namespace := &corev1.Namespace{}
+						return apierrors.ReasonForError(testContext.ClustersClients[remoteClusterID].Get(ctx,
+							types.NamespacedName{Name: testNamespaceName}, namespace))
+					}, shortTimeout, interval).Should(Equal(metav1.StatusReasonNotFound))
+				}
+
+			}
+
+		})
+
+		It("Delete the local namespace and check if the remote namespaces are deleted", func() {
+			By(" 1 - Getting the local namespace and delete it")
+			namespace := &corev1.Namespace{}
+			Eventually(func() metav1.StatusReason {
+				err := testContext.Clusters[localIndex].ControllerClient.Get(ctx, types.NamespacedName{Name: testNamespaceName}, namespace)
+				_ = testContext.Clusters[localIndex].ControllerClient.Delete(ctx, namespace)
+				return apierrors.ReasonForError(err)
+			}, longTimeout, interval).Should(Equal(metav1.StatusReasonNotFound))
```

Comment:
IMHO they are mostly equivalent, since deleting the namespace automatically leads to the deletion of the NamespaceOffloading. The other alternative was just a bit more explicit, but the current approach is also perfectly fine to me

Reply:
Yes you are right it is more explicit